### PR TITLE
Use CONTAINERS_CONF cgroups flag for remote API.

### PR DESCRIPTION
**Pull request by rhatdan**

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixed: #12550

**Comment by mheon** [comment]
I got a bit worried about the diffstat, but CGroup -> Cgroup makes sense to me

**Comment by rhatdan** [comment]
Yes I went both ways and this seems least breaking.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Comment by jwhonce** [comment]
LGTM

File: test/e2e/containers_conf_test.go [hunk 1]
```diff
@@ -484,4 +484,35 @@ var _ = Describe("Podman run", func() {
...
+
+		result := podmanTest.Podman([]string{"create", ALPINE, "true"})
+		result.WaitWithDefaultTimeout()
+		Expect(result).Should(Exit(0))
+
+		inspect := podmanTest.Podman([]string{"inspect", "--format", "{{ .HostConfig.Cgroups }}", result.OutputToString()})
+		inspect.WaitWithDefaultTimeout()
+		Expect(inspect.OutputToString()).To(Not(Equal("disabled")))
+
+		os.Setenv("CONTAINERS_CONF", conffile)
```
Comment (vrothberg) [inline test/e2e/containers_conf_test.go:505]: Can you unset it in a `defer`?

**Review by vrothberg — COMMENTED** [review]
Regarding `CGroup` vs `Cgroup`. I'd prefer to call it `cgroup` in error messages since that's what man pages etc. use. The camel cases in the _code_ LGTM.

File: test/e2e/containers_conf_test.go (hunk 1 above)
Comment (rhatdan) [reply test/e2e/containers_conf_test.go:505]: Not needed in containers_conf_test.  The Before/After are taking care of this.

**Pull request merged**